Packs/CommonScripts/Scripts/FormatURL/FormatURL.py

```diff
@@ -1,291 +1,549 @@
-    return url_
-
-
-def replace_protocol(url_: str) -> str:
-    """
-    Replaces URL protocol with expected protocol. Examples can be found in tests.
-    Args:
-        url_ (str): URL to replace the protocol by the given examples above.
+    def __str__(self):
+        return (
+            f'Scheme = {self.scheme}\nUser_info = {self.user_info}\nHostname = {self.hostname}\nPort = {self.port}\n'
+            f'Path = {self.path}\nQuery = {self.query}\nFragment = {self.fragment}')
 
-    Returns:
-        (str): URL with replaced protocol, if needed to replace, else the URL itself.
-    """
-    for prefix_to_normalize in PREFIX_TO_NORMALIZE:
-        if url_.startswith(prefix_to_normalize):
-            url_ = url_.replace(prefix_to_normalize, HTTP)
-    lowercase_url = url_.lower()
-    for starts_with, does_not_start_with, to_replace in PREFIX_CHANGES:
-        if lowercase_url.startswith(starts_with) and (
-                not does_not_start_with or not lowercase_url.startswith(does_not_start_with)):
-            url_ = url_.replace(starts_with, to_replace)
-    if url_.startswith('http:') and not url_.startswith('http:/'):
-        url_ = url_.replace('http:', 'http://')
-    if url_.startswith('https:') and not url_.startswith('https:/'):
-        url_ = url_.replace('https:', 'https://')
-    return url_
-
-
-def remove_special_chars_from_start_and_end_of_url(url_: str) -> str:
+
+class URLCheck(object):
     """
-    Removes brackets and quotes characters from the beginning and the end of the URL if there are any.
-    Args:
-        url_ (str): URL to remove the special characters from.
-    Returns:
-        (str): URL with characters brackets, if needed to remove, else the URL itself.
+    This class will build and validate a URL based on "URL Living Standard" (https://url.spec.whatwg.org)
     """
 
-    def is_ipv6(url: str) -> bool:
+    def __init__(self, original_url):
+        self.modified_url = original_url
+        self.original_url = original_url
+        self.url = URLType(original_url)
+        self.base = 0
+        self.output = ''
+
+        self.inside_brackets = False
+        self.port = False
+        self.query = False
+        self.fragment = False
+        self.done = False
+
+        self.sub_delims = ("!", "$", "&", "'", "(", ")", "*", "+", ",", ";", "=")
+        self.brackets = ("\"", "'", "[", "]", "{", "}", "(", ")", ",")
+        self.url_code_points = ("\u0021", "\u0024", "\u0026", "\u0027", "\u0028", "\u0029", "\u002A", "\u002B",
+                                "\u002C", "\u002D", "\u002E", "\u002F", "\u003A", "\u003B", "\u003D", "\u003F",
+                                "\u0040", "\u005F", "\u007E")
+
+        if self.original_url:
+            self.remove_leading_chars()
+
+        else:
+            raise URLError("Empty string given")
+
+        if "//" in self.modified_url[:8]:
+            """
+            The URL seems to have a scheme indicated by presence of "//"
+            """
```

Comment:
For inner-code comments you should use a normal comment, i.e. `#`, doctstring should usually be used in functions/classes/modules.
```suggestion
            # The URL seems to have a scheme indicated by presence of "//"
```